### ISSUE TITLE
api.appcenter.ms invalid SSL certificate when requesting without "server_name" field explicitly set

### ISSUE THREAD

**Issue by michaljrk** [microsoft/appcenter#2070]
The issue got revealed when App Center Jenkins plugin has stopped working due to certificate validation errors. 
It turns out to be related to api.appcenter.ms endpoint configuration changes and SNI, which result in presenting a "fake" Kubernetes Ingress certificate when "server_name" is not specified explicitly in the request.

**Reproduction steps from openssl:**

- request **without** specifying the "servername" flag explicitly:

`openssl s_client -showcerts -connect api.appcenter.ms:443`

"fake" Kubernetes Ingress certificate in the output:

```Server certificate
subject=/O=Acme Co/CN=Kubernetes Ingress Controller Fake Certificate
issuer=/O=Acme Co/CN=Kubernetes Ingress Controller Fake Certificate
---
No client certificate CA names sent
---
SSL handshake has read 1391 bytes and written 421 bytes
---
New, TLSv1/SSLv3, Cipher is ECDHE-RSA-AES128-GCM-SHA256
Server public key is 2048 bit
Secure Renegotiation IS supported
Compression: NONE
Expansion: NONE
SSL-Session:
    Protocol  : TLSv1.2
    Cipher    : ECDHE-RSA-AES128-GCM-SHA256
    Session-ID: 2D139F6A03B9FAF75A61FD854B03CB24CC646C25DBEC66D3808B2BE8E9512FCC
    Session-ID-ctx:
    Master-Key: A4765DD8043CDE8F44B3633E0DB2B78C32073059E232B7F6116622A9F0A0455E71F0AA4A2AC8507F9A346778F275A2AA
    Key-Arg   : None
    PSK identity: None
    PSK identity hint: None
    SRP username: None
    Start Time: 1607349341
    Timeout   : 300 (sec)
    Verify return code: 21 (unable to verify the first certificate)
```

- request with the "servername" flag explicitly set:

`openssl s_client -showcerts -servername api.appcenter.ms -connect api.appcenter.ms:443`

result (valid certificate):

```
---
No client certificate CA names sent
---
SSL handshake has read 4945 bytes and written 446 bytes
---
New, TLSv1/SSLv3, Cipher is ECDHE-RSA-AES128-GCM-SHA256
Server public key is 2048 bit
Secure Renegotiation IS supported
Compression: NONE
Expansion: NONE
SSL-Session:
    Protocol  : TLSv1.2
    Cipher    : ECDHE-RSA-AES128-GCM-SHA256
    Session-ID: 91F7E6F2F13056FF416FB09864FB3331CCF63D43CC07E68B2C5682C897398E5F
    Session-ID-ctx:
    Master-Key: 03487026B3BAF22CA42FD4F037ACD7CA4C132A78767097633E7D0D2642D32EEB0C3D7470369AD569D361674A9893BA4F
    Key-Arg   : None
    PSK identity: None
    PSK identity hint: None
    SRP username: None
    Start Time: 1607349413
    Timeout   : 300 (sec)
    Verify return code: 19 (self signed certificate in certificate chain)
```

**Solution proposal:**

Present a valid SSL certificate even when requesting client doesn't set the "servername" field explicitly.

**Comment by yevgeniy-logachev** [comment]
I've got same issue with uploading from Jenkins. Is there any workaround?

**Comment by michaljrk** [comment]
Not any that I'm aware of @yevgeniy-logachev. 
I tried adding the "fake" certificate to Jenkins key store, but then realized that there are different "fake" ssl certs presented every time I send the request.

**Comment by matthiaswenz** [comment]
Thanks for reaching out to us - we are looking into this and what might have changed here in the certificate(s) offered by api.appcenter.ms.

**Comment by matthiaswenz** [comment]
@michaljrk Is this something that changed lately, with a specific version of the Jenkins plugin?
We don't see any recent changes to our configuration around which certificates we present - so this might be more likely an issue of the HTTPS/TLS client in the Jenkins plugin not having SNI support enabled.

I suppose you are referring to this plugin https://github.com/jenkinsci/appcenter-plugin, correct? This is a plugin provided by the community not App Center itself, so I would recommend to check any issues with the owners of the plugin and if they are aware of this.

**Comment by michaljrk** [comment]
That's exactly the plugin you referenced @matthiaswenz . I did open a JiRA ticket for the plugin maintainers first (https://issues.jenkins.io/browse/JENKINS-64381), then, after not seeing much interest, I decided to create this GH Issue too.

> We don't see any recent changes to our configuration around which certificates we present - so this might be more likely an issue of the HTTPS/TLS client in the Jenkins plugin not having SNI support enabled.

That's probably the root cause, but what makes me think is that it stopped working without introducing any changes for the plugin (or updating it to a newer version), which lead me to a conclusion that it could be some api.appcenter.ms endpoint/ssl/Kubernetes Ingress reconfiguration that introduced it. 
Also, as shown in my `openssl` steps, the endpoint is presenting self signed kubernetes Ingress certificates, and that was not the case before, I believe.
